pallets/schemas/src/lib.rs

```diff
@@ -524,7 +821,38 @@ pub mod pallet {
 					payload_location == PayloadLocation::Itemized,
 				Error::<T>::InvalidSetting
 			);
-			Self::add_schema(model, model_type, payload_location, settings)
+			let schema_name = match optional_schema_name {
+				None => None,
+				Some(name_payload) => {
+					let parsed_name = SchemaName::try_parse::<T>(name_payload, true)?;
+					Some(parsed_name)
+				},
+			};
+			let schema_id = Self::add_schema(
+				model,
+				model_type,
+				payload_location,
+				settings,
+				schema_name.clone(),
+			)?;
+			Ok((schema_id, schema_name))
+		}
+
+		/// a method to return all versions of a schema name with their schemaIds
+		/// Warning: Must only get called from RPC, since the number of DB accesses is not deterministic
+		pub fn get_schema_versions(schema_name: Vec<u8>) -> Option<Vec<SchemaVersionResponse>> {
+			let bounded_name = BoundedVec::try_from(schema_name).ok()?;
```

Comment:
From my understanding since this is an RPC call which only gets executes on the receiving node I don't think it makes any difference in terms of optimization